heartbeat/process/sender.go

```diff
@@ -89,20 +95,28 @@ func NewSender(arg ArgHeartbeatSender) (*Sender, error) {
 		nodeDisplayName:      arg.NodeDisplayName,
 		keyBaseIdentity:      arg.KeyBaseIdentity,
 		hardforkTrigger:      arg.HardforkTrigger,
+		chainHandler:         arg.ChainHandler,
 	}
 
 	return sender, nil
 }
 
 // SendHeartbeat broadcasts a new heartbeat message
 func (s *Sender) SendHeartbeat() error {
-	hb := &data.Heartbeat{
+	nonce := uint64(0)
+	crtBlock := s.chainHandler.GetCurrentBlockHeader()
+	if !check.IfNil(crtBlock) {
```

Comment:
blockchain hook would give you the current block nonce directly.

Reply:
This was easier to inject from the Node struct, will remain as it is.